### ISSUE TITLE
Rails 3.2.1 does not start with rspec-rails 3.2.2

### ISSUE THREAD

**Issue by tagliala** [rspec/rspec-rails#1386]
It should be a consequence of #1327, as per https://github.com/rspec/rspec-rails/pull/1327#issuecomment-108278230

```sh
$ rails c
~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:438:in `method_missing': undefined method `preview_path=' for ActionMailer::Base:Class (NoMethodError)
```

**Comment by cupakromer** [comment]
I'm looking into this now.

**Comment by myronmarston** [comment]
Thanks for hopping on it so quickly, @cupakromer.  I'll hold off on the 3.3 release until this is done (I think everything else on our 3.3 list already got done).

We should probably backport the fix to 3.2.3 as well, and I'll be happy to release that, too.

**Comment by cupakromer** [comment]
I think I see the issue but I've been unable to reproduce the error. I'll need to be able to reproduce before I can ensure I've fixed the issue.

@tagliala if you could please update this issue with the following I'd really appreciate it:

- full stack trace
- output of `ruby -v`
- output of `rails -v`

**Comment by cupakromer** [comment]
also, @olance if you could provide additional details I would appreciate it

**Comment by olance** [comment]
Sure! There you go:

```
~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:438:in `method_missing': undefined method `preview_path=' for ActionMailer::Base:Class (NoMethodError)
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `block (3 levels) in <class:Railtie>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `each'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `block (2 levels) in <class:Railtie>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:36:in `instance_eval'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:43:in `block in run_load_hooks'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:42:in `each'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:42:in `run_load_hooks'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:746:in `<class:Base>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:326:in `<module:ActionMailer>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:11:in `<top (required)>'
     from ~/dev/bmb_front/config/initializers/email.rb:2:in `<top (required)>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `load'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `block in load'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `load'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/engine.rb:593:in `block (2 levels) in <class:Engine>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/engine.rb:592:in `each'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/engine.rb:592:in `block in <class:Engine>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/initializable.rb:30:in `instance_exec'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/initializable.rb:30:in `run'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/initializable.rb:55:in `block in run_initializers'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/initializable.rb:54:in `each'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/initializable.rb:54:in `run_initializers'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/application.rb:136:in `initialize!'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/railties-3.2.21/lib/rails/railtie/configurable.rb:30:in `method_missing'
     from ~/dev/bmb_front/config/environment.rb:5:in `<top (required)>'
     from config.ru:4:in `require'
     from config.ru:4:in `block in <main>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rack-1.4.5/lib/rack/builder.rb:51:in `instance_eval'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rack-1.4.5/lib/rack/builder.rb:51:in `initialize'
     from config.ru:1:in `new'
     from config.ru:1:in `<main>'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/lib/unicorn.rb:48:in `eval'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/lib/unicorn.rb:48:in `block in builder'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/lib/unicorn/http_server.rb:768:in `call'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/lib/unicorn/http_server.rb:768:in `build_app!'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/lib/unicorn/http_server.rb:137:in `start'
     from ~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/unicorn-4.9.0/bin/unicorn:126:in `<top (required)>'
     from ~/.rbenv/versions/2.1.2/bin/unicorn:23:in `load'
     from ~/.rbenv/versions/2.1.2/bin/unicorn:23:in `<main>'
```

```
$ bundle exec rails -v
Rails 3.2.21
```

```
$ bundle exec ruby -v 
ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-darwin14.0]
```

**Comment by olance** [comment]
From my debug session, I have found that around line 34 in `~/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb`, `config.action_mailer` already contained a value for `:preview_path`. It was set to a subdirectory of my project's `specs` dir.

I searched for `preview_path` in project's and my gems' directory and it was only found in rspec-rails.

**Comment by cupakromer** [comment]
What's in `~/dev/bmb_front/config/initializers/email.rb`? I see that is part of the path. This does help shed a little more light on the order of operations.

**Comment by cupakromer** [comment]
Ok, I've got it reproduced locally. The config is happy to take the option. The error only occurs once `ActionMailer::Base` is loaded. I'll start working on a fix with this assumption being what triggers your issue as well.

**Comment by tagliala** [comment]
```
$ rails c
~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/base.rb:438:in `method_missing': undefined method `preview_path=' for ActionMailer::Base:Class (NoMethodError)
	from ~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `block (3 levels) in <class:Railtie>'
	from ~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `each'
	from ~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:34:in `block (2 levels) in <class:Railtie>'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:36:in `instance_eval'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:26:in `block in on_load'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:25:in `each'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/lazy_load_hooks.rb:25:in `on_load'
	from ~/.rvm/gems/ruby-2.1.5/gems/actionmailer-3.2.21/lib/action_mailer/railtie.rb:26:in `block in <class:Railtie>'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/initializable.rb:30:in `instance_exec'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/initializable.rb:30:in `run'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/initializable.rb:55:in `block in run_initializers'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/initializable.rb:54:in `each'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/initializable.rb:54:in `run_initializers'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/application.rb:136:in `initialize!'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/railtie/configurable.rb:30:in `method_missing'
	from ~/dev/project/config/environment.rb:5:in `<top (required)>'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `block in require'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
	from ~/.rvm/gems/ruby-2.1.5/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/application.rb:103:in `require_environment!'
	from ~/.rvm/gems/ruby-2.1.5/gems/railties-3.2.21/lib/rails/commands.rb:40:in `<top (required)>'
	from script/rails:6:in `require'
	from script/rails:6:in `<main>'
```

```
$ rails -v
Rails 3.2.21
```

```
$ ruby -v
ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin14.0]
```

environment.rb
```
# Load the rails application
require File.expand_path('../application', __FILE__)

# Initialize the rails application
Project::Application.initialize!
```

**Comment by cupakromer** [comment]
@tagliala @olance thank you both for reporting this :blue_heart: I'm very sorry for causing this regression. I'll try to get the fix up by the end of the day.

**Comment by olance** [comment]
No worries! Thanks for fixing it so quickly!

**Comment by tagliala** [comment]
@cupakromer thank you! no problem here :+1:

**Comment by myronmarston** [comment]
I just released 3.2.3 with @cupakromer's fix.

**Comment by tagliala** [comment]
thanks!
